From a42d3834f288c07d20c9192bf818023627993334 Mon Sep 17 00:00:00 2001
From: "Unknown W. Brackets" <checkins@unknownbrackets.org>
Date: Sat, 20 Mar 2021 17:06:21 -0700
Subject: [PATCH] Build: Remove "fake" vertex decoder jit.

We don't really need this.  Just replace it with a stub that always fails
to generate a vertex decoder.
---
 CMakeLists.txt                      |   2 -
 GPU/Common/VertexDecoderCommon.h    |  28 +--
 GPU/Common/VertexDecoderFake.cpp    | 266 ----------------------------
 UWP/GPU_UWP/GPU_UWP.vcxproj         |   1 -
 UWP/GPU_UWP/GPU_UWP.vcxproj.filters |   1 -
 5 files changed, 18 insertions(+), 280 deletions(-)
 delete mode 100644 GPU/Common/VertexDecoderFake.cpp

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 598107a6b..ce5a6a9c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -423,7 +423,6 @@ source_group(MIPS FILES ${CommonMIPS})
 if(NOT (X86 OR ARM OR MIPS))
 	set(CommonFake
 		Common/FakeCPUDetect.cpp
-		Common/FakeEmitter.h
 	)
 	source_group(Fake FILES ${CommonFake})
 endif()
@@ -1251,7 +1250,6 @@ list(APPEND CoreExtra
 list(APPEND CoreExtra
 	Core/MIPS/MIPS/MipsJit.cpp
 	Core/MIPS/MIPS/MipsJit.h
-	GPU/Common/VertexDecoderFake.cpp
 )
 
 if(NOT MOBILE_DEVICE)
diff --git a/GPU/Common/VertexDecoderCommon.h b/GPU/Common/VertexDecoderCommon.h
index 9377b3611..2911e8e6f 100644
--- a/GPU/Common/VertexDecoderCommon.h
+++ b/GPU/Common/VertexDecoderCommon.h
@@ -35,8 +35,6 @@
 #include "Common/Arm64Emitter.h"
 #elif PPSSPP_ARCH(X86) || PPSSPP_ARCH(AMD64)
 #include "Common/x64Emitter.h"
-#elif PPSSPP_ARCH(MIPS)
-#include "Common/MipsEmitter.h"
 #else
 #include "Common/FakeEmitter.h"
 #endif
@@ -601,18 +599,17 @@ class VertexDecoder {
 //
 // that's it!
 
-
 #if PPSSPP_ARCH(ARM)
-class VertexDecoderJitCache : public ArmGen::ARMXCodeBlock {
+#define VERTEXDECODER_JIT_BACKEND ArmGen::ARMXCodeBlock
 #elif PPSSPP_ARCH(ARM64)
-class VertexDecoderJitCache : public Arm64Gen::ARM64CodeBlock {
+#define VERTEXDECODER_JIT_BACKEND Arm64Gen::ARM64CodeBlock
 #elif PPSSPP_ARCH(X86) || PPSSPP_ARCH(AMD64)
-class VertexDecoderJitCache : public Gen::XCodeBlock {
-#elif PPSSPP_ARCH(MIPS)
-class VertexDecoderJitCache : public MIPSGen::MIPSCodeBlock {
-#else
-class VertexDecoderJitCache : public FakeGen::FakeXCodeBlock {
+#define VERTEXDECODER_JIT_BACKEND Gen::XCodeBlock
 #endif
+
+
+#ifdef VERTEXDECODER_JIT_BACKEND
+class VertexDecoderJitCache : public VERTEXDECODER_JIT_BACKEND {
 public:
 	VertexDecoderJitCache();
 
@@ -705,3 +702,14 @@ class VertexDecoderJitCache : public FakeGen::FakeXCodeBlock {
 	Arm64Gen::ARM64FloatEmitter fp;
 #endif
 };
+#else
+class VertexDecoderJitCache : public FakeGen::FakeXCodeBlock {
+public:
+	VertexDecoderJitCache();
+
+	JittedVertexDecoder Compile(const VertexDecoder &dec, int32_t *jittedSize) {
+		return nullptr;
+	}
+	void Clear();
+};
+#endif
-- 
2.30.0

